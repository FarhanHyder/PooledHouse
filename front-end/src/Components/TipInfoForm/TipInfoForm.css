#add {
    margin: 5px 5px 5px 5px;
}