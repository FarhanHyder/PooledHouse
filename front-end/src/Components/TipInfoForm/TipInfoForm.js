import React from 'react';
import {Col, Row, Container} from 'react-bootstrap';
import {Form, Button} from 'react-bootstrap';

import { API, graphqlOperation, Auth } from "aws-amplify";
import * as mutations from '../../graphql/mutations';

class TipInfoForm extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            user: '',
            shift_length: '',
            shift_time: 'AM',
            shift_date: '',
            shift_position: 'Bartender',
            takehome: '',
            business_name: '',
            business_street_address: '',
            business_city: '',
            business_state: '', 
            business_zip: '',
            neighborhood: '',
        }
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChange = (e) => {
        this.setState({ [e.target.name]: e.target.value })
    }

    async handleSubmit(e) {
        e.preventDefault();
        let t_e = {
            id: (Math.floor(Math.random() * 100000)),
            user: this.state.user,
            shift_length: this.state.shift_length,
            shift_time: this.state.shift_time,
            shift_date: this.state.shift_date,
            shift_position: this.state.shift_position,
            takehome: this.state.takehome,
            business_name: this.state.business_name,
            business_street_address: this.state.business_street_address,
            business_city: this.state.business_city,
            business_state: this.state.business_state,
            business_zip: this.state.business_zip,
            neighborhood: this.state.neighborhood,
        }
        console.log(t_e);
        try {
            await API.graphql(graphqlOperation(mutations.createTipEntry, {input: t_e}));
            console.log(t_e);
            alert("Success!");
            this.setState({
                user: '',
                shift_length: '',
                shift_time: 'AM',
                shift_date: '',
                shift_position: 'Bartender',
                takehome: '',
                business_name: '',
                business_street_address: '',
                business_city: '', 
                business_state: '',
                business_zip: '',
                neighborhood: ''
            })
        } catch (err) {
            console.log('error creating tipentry: ', err);
            alert("tipentry failed");
        }
    }

    async componentDidMount() {
        let current_user = await Auth.currentAuthenticatedUser();
        let un = current_user.username;
        this.setState({
          user: un
        })
    }

    render () {

        const neighborhoods = [ 'Battery Park City', 'Central Harlem', 'Central Park',
                                'Chelsea', 'Chinatown', 'City Hall Area', 'East Village',
                                'Ellis Island', 'Financial District', 'Flatiron District',
                                'Governors Island', 'Gramercy', 'Greenwich Village', 'Hamilton Heights',
                                "Hell's Kitchen", 'Inwood', 'Kips Bay', 'Liberty Island', 'Little Italy',
                                'Lower East Side', 'Marble Hill', 'Midtown East', 'Midtown West', 'Morningside Heights',
                                'Murray Hill', 'NoHo', 'Nolita', "Randall's Island", 'Roosevelt Island', 
                                'SoHo', 'Spanish Harlem', 'Stuyvesant Town', 'Tribeca', 'Two Bridges', 'Upper East Side',
                                'Upper West Side', 'Washington Heights', 'West Village'
                               ];
        
        let hoodlist = [];

        neighborhoods.forEach(function(neighborhood) {
            hoodlist.push({ label: neighborhood, value: neighborhood })
        })

        const { shift_length, 
                shift_time,
                shift_date,
                shift_position,
                takehome,
                business_name,
                business_street_address,
                business_city,
                business_state,
                business_zip,
                neighborhood
            } = this.state;

        return (
            <div>
        <Container fluid='true'>
            <Form className="text-left newTips">
                <Row>
                    <Col>
                        <Form.Group controlID="shift_date">
                                <Form.Label>Shift Date</Form.Label>
                                <Form.Control
                                    type='date'
                                    name="shift_date"
                                    value={shift_date}
                                    onChange={this.handleChange}
                                />
                            </Form.Group> 
                    </Col>
                    <Col>
                        <Form.Group controlId="shift_time">
                            <Form.Label>Shift Time</Form.Label>
                            <Form.Control 
                                    as='select' 
                                    name="shift_time" 
                                    value={shift_time}
                                    onChange={this.handleChange}>
                                    <option>AM</option>
                                    <option>PM</option>
                                </Form.Control>
                        </Form.Group>
                    </Col>
                    <Col>
                        <Form.Group controlID="shift_length">
                            <Form.Label>Shift Length (hours)</Form.Label>
                            <Form.Control
                                type='number'
                                name="shift_length"
                                value={shift_length}
                                onChange={this.handleChange}
                            />
                        </Form.Group>
                    </Col>
                    <Col>
                        <Form.Group controlID="shift_position">
                                <Form.Label>Position</Form.Label>
                                <Form.Control 
                                    as='select' 
                                    name="shift_position" 
                                    value={shift_position}
                                    onChange={this.handleChange}>
                                    <option>Bartender</option>
                                    <option>Server</option>
                                    <option>Barback</option>
                                    <option>Busser</option>
                                    <option>Other</option>
                                </Form.Control>
                        </Form.Group>
                    </Col>
                </Row>
                <Row>
                    <Col>
                        <Form.Group controlID="takehome">
                            <Form.Label>Tip Takehome</Form.Label>
                            <Form.Control
                                type='number'
                                name="takehome"
                                value={takehome}
                                onChange={this.handleChange}
                            />
                        </Form.Group>
                    </Col>
                    <Col>
                        <Form.Group controlID="business_name">
                            <Form.Label>Business Name</Form.Label>
                            <Form.Control
                                type='text'
                                placeholder="i.e. Hot Jim's BBQ"
                                name="business_name"
                                value={business_name}
                                onChange={this.handleChange}
                            />
                        </Form.Group>
                    </Col>
                    <Col>
                        <Form.Group controlID="business_street_address">
                            <Form.Label>Street Address</Form.Label>
                            <Form.Control
                                type='text'
                                placeholder="i.e. 123 E 9th St"
                                name="business_street_address"
                                value={business_street_address}
                                onChange={this.handleChange}
                            />
                        </Form.Group>
                    </Col>
                    <Col>
                        <Form.Group controlID="business_city">
                            <Form.Label>City</Form.Label>
                            <Form.Control
                                type='text'
                                placeholder="i.e. New York City"
                                name="business_city"
                                value={business_city}
                                onChange={this.handleChange}
                            />
                        </Form.Group>
                    </Col>
                    <Col>
                        <Form.Group controlID="business_state">
                            <Form.Label>State</Form.Label>
                            <Form.Control
                                type='text'
                                placeholder="i.e. NY"
                                name="business_state"
                                value={business_state}
                                onChange={this.handleChange}
                            />
                        </Form.Group>
                    </Col>
                    <Col>
                        <Form.Group controlID="business_zip">
                            <Form.Label>Zipcode</Form.Label>
                            <Form.Control
                                type='text'
                                placeholder="i.e. 11105"
                                name="business_zip"
                                value={business_zip}
                                onChange={this.handleChange}
                            />
                        </Form.Group>
                    </Col>
                    <Col>
                        <Form.Group controlID="neighborhood">
                                <Form.Label>Neighborhood</Form.Label>
                                <Form.Control 
                                    as='select' 
                                    name="neighborhood" 
                                    value={neighborhood}
                                    onChange={this.handleChange}>
                                    { neighborhoods.map((neighborhood, index) => {
                                        return (
                                            <option key={index} value={neighborhood}>{neighborhood}</option>
                                        )
                                    })}
                                </Form.Control>
                                <Button 
                                variant="primary" 
                                type="submit" 
                                onClick={(ev) => this.handleSubmit(ev)}>Add Tips
                                </Button>
                        </Form.Group>
                    </Col>
                </Row>
            </Form>
        </Container>
        </div>
        );
    }
}

export default TipInfoForm;
