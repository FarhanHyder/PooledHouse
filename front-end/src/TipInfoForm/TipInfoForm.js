import React from 'react';
import {Col, Row, Container} from 'react-bootstrap';
import {Form, Button, ButtonToolbar, Navbar} from 'react-bootstrap';

class TipInfoForm extends React.Component {
    constructor(props) {
        super(props);
        //this.submit = this.submit.bind(this);
        this.state = {
            shift_length: '',
            shift_time: '',
            shift_date: '',
            position: 'Bartender',
            takehome: '',
            business_name: '',
            business_address: '',
        }
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChange = (e) => {
        this.setState({ [e.target.name]: e.target.value })
    }

    handleSubmit = (e) => {
        alert("submitted");
    }

    render () {
        const { shift_length, 
                shift_time,
                shift_date,
                position,
                takehome,
                business_name,
                business_address
            } = this.state;

        return (
            <div>
        <Navbar className="bg-olive justify-content-between">
            <h3 className="text-color-white">Pooled House</h3>
            <ButtonToolbar>
                <Button href="#" variant="link" className="text-color-white" onClick={this.props.handler}>Home</Button>
            </ButtonToolbar>
        </Navbar>
        <Container fluid='true'>
            <Form className="text-left newTips" onSubmit={this.handleSubmit}>
                <Row>
                    <Col>
                        <Form.Group controlID="shift_date">
                                <Form.Label>Shift Date</Form.Label>
                                <Form.Control
                                    type='date'
                                    name="shift_date"
                                    value={shift_date}
                                    onChange={this.handleChange}
                                />
                            </Form.Group> 
                    </Col>
                    <Col>
                        <Form.Group controlId="shift_time">
                            <Form.Label>Shift Time</Form.Label>
                            { /*this div just helps with formatting */}
                            <div>
                                <Form.Check 
                                    inline label='AM' 
                                    type='radio' 
                                    name='shift_time' 
                                    id='radio-AM'
                                    value={shift_time}
                                    onChange={this.handleChange} />
                                <Form.Check 
                                    inline label='PM' 
                                    type='radio' 
                                    name='shift_time' 
                                    id='radio-PM'
                                    value={shift_time}
                                    onChange={this.handleChange} />
                            </div>
                        </Form.Group>
                    </Col>
                    <Col>
                        <Form.Group controlID="shift_length">
                            <Form.Label>Shift Length (hours)</Form.Label>
                            <Form.Control
                                type='number'
                                name="shift_length"
                                value={shift_length}
                                onChange={this.handleChange}
                            />
                        </Form.Group>
                    </Col>
                    <Col>
                        <Form.Group controlID="position">
                                <Form.Label>Position</Form.Label>
                                <Form.Control 
                                    as='select' 
                                    name="position" 
                                    value={position}
                                    onChange={this.handleChange}>
                                    <option>Bartender</option>
                                    <option>Server</option>
                                    <option>Barback</option>
                                    <option>Busser</option>
                                    <option>Other</option>
                                </Form.Control>
                        </Form.Group>
                    </Col>
                </Row>
                <Row>
                    <Col>
                        <Form.Group controlID="takehome">
                            <Form.Label>Tip Takehome</Form.Label>
                            <Form.Control
                                type='number'
                                name="takehome"
                                value={takehome}
                                onChange={this.handleChange}
                            />
                        </Form.Group>
                    </Col>
                    <Col>
                        <Form.Group controlID="business_name">
                            <Form.Label>Business Name</Form.Label>
                            <Form.Control
                                type='text'
                                placeholder="i.e. Hot Jim's BBQ"
                                name="business_name"
                                value={business_name}
                                onChange={this.handleChange}
                            />
                        </Form.Group>
                    </Col>
                    <Col>
                        <Form.Group controlID="business_address">
                            <Form.Label>Business Address</Form.Label>
                            <Form.Control
                                type='text'
                                placeholder="i.e. 123 E 9th St, New York, NY 99999"
                                name="business_address"
                                value={business_address}
                                onChange={this.handleChange}
                            />
                            <Button variant="primary" type="submit">Submit</Button>
                        </Form.Group>
                    </Col>
                </Row>
            </Form>
        </Container>
        </div>
        );
    }
}

export default TipInfoForm;
