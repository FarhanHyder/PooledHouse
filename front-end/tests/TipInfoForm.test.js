import React from 'react';
import renderer from 'react-test-renderer';
import TipInfoForm from '../src/Components/TipInfoForm/TipInfoForm';

test('TipInfoForm component snapshot test', () => {
  const thisProps = {
    'shift_length': 5,
    'shift_time': 'PM',
    'shift_date': '04/30/2019',
    'shift_position': 'Busser',
    'takehome': 87,
    'business_name': 'Poolinos',
    'business_street_address': '292 Bowery',
    'business_city': 'New York',
    'business_state': 'NY',
    'business_zip': '10012',
    'neighborhood': 'Soho'
  };
  const component = renderer.create(
    <TipInfoForm props={thisProps} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});