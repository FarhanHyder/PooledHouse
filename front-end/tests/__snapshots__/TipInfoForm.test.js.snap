// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TipInfoForm component snapshot test 1`] = `
<div>
  <div
    className="container-fluid"
  >
    <form
      className="text-left newTips"
    >
      <div
        className="row"
      >
        <div
          className="col"
        >
          <div
            className="form-group"
            controlID="shift_date"
          >
            <label
              className="form-label"
            >
              Shift Date
            </label>
            <input
              className="form-control"
              name="shift_date"
              onChange={[Function]}
              type="date"
              value=""
            />
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="form-group"
          >
            <label
              className="form-label"
              htmlFor="shift_time"
            >
              Shift Time
            </label>
            <select
              className="form-control"
              id="shift_time"
              name="shift_time"
              onChange={[Function]}
              value="AM"
            >
              <option>
                AM
              </option>
              <option>
                PM
              </option>
            </select>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="form-group"
            controlID="shift_length"
          >
            <label
              className="form-label"
            >
              Shift Length (hours)
            </label>
            <input
              className="form-control"
              name="shift_length"
              onChange={[Function]}
              type="number"
              value=""
            />
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="form-group"
            controlID="shift_position"
          >
            <label
              className="form-label"
            >
              Position
            </label>
            <select
              className="form-control"
              name="shift_position"
              onChange={[Function]}
              value="Bartender"
            >
              <option>
                Bartender
              </option>
              <option>
                Server
              </option>
              <option>
                Barback
              </option>
              <option>
                Busser
              </option>
              <option>
                Other
              </option>
            </select>
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col"
        >
          <div
            className="form-group"
            controlID="takehome"
          >
            <label
              className="form-label"
            >
              Tip Takehome
            </label>
            <input
              className="form-control"
              name="takehome"
              onChange={[Function]}
              type="number"
              value=""
            />
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="form-group"
            controlID="business_name"
          >
            <label
              className="form-label"
            >
              Business Name
            </label>
            <input
              className="form-control"
              name="business_name"
              onChange={[Function]}
              placeholder="i.e. Hot Jim's BBQ"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="form-group"
            controlID="business_street_address"
          >
            <label
              className="form-label"
            >
              Street Address
            </label>
            <input
              className="form-control"
              name="business_street_address"
              onChange={[Function]}
              placeholder="i.e. 123 E 9th St"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="form-group"
            controlID="business_city"
          >
            <label
              className="form-label"
            >
              City
            </label>
            <input
              className="form-control"
              name="business_city"
              onChange={[Function]}
              placeholder="i.e. New York City"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="form-group"
            controlID="business_state"
          >
            <label
              className="form-label"
            >
              State
            </label>
            <input
              className="form-control"
              name="business_state"
              onChange={[Function]}
              placeholder="i.e. NY"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="form-group"
            controlID="business_zip"
          >
            <label
              className="form-label"
            >
              Zipcode
            </label>
            <input
              className="form-control"
              name="business_zip"
              onChange={[Function]}
              placeholder="i.e. 11105"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="form-group"
            controlID="neighborhood"
          >
            <label
              className="form-label"
            >
              Neighborhood
            </label>
            <select
              className="form-control"
              name="neighborhood"
              onChange={[Function]}
              value=""
            >
              <option
                value="Battery Park City"
              >
                Battery Park City
              </option>
              <option
                value="Central Harlem"
              >
                Central Harlem
              </option>
              <option
                value="Central Park"
              >
                Central Park
              </option>
              <option
                value="Chelsea"
              >
                Chelsea
              </option>
              <option
                value="Chinatown"
              >
                Chinatown
              </option>
              <option
                value="City Hall Area"
              >
                City Hall Area
              </option>
              <option
                value="East Village"
              >
                East Village
              </option>
              <option
                value="Ellis Island"
              >
                Ellis Island
              </option>
              <option
                value="Financial District"
              >
                Financial District
              </option>
              <option
                value="Flatiron District"
              >
                Flatiron District
              </option>
              <option
                value="Governors Island"
              >
                Governors Island
              </option>
              <option
                value="Gramercy"
              >
                Gramercy
              </option>
              <option
                value="Greenwich Village"
              >
                Greenwich Village
              </option>
              <option
                value="Hamilton Heights"
              >
                Hamilton Heights
              </option>
              <option
                value="Hell's Kitchen"
              >
                Hell's Kitchen
              </option>
              <option
                value="Inwood"
              >
                Inwood
              </option>
              <option
                value="Kips Bay"
              >
                Kips Bay
              </option>
              <option
                value="Liberty Island"
              >
                Liberty Island
              </option>
              <option
                value="Little Italy"
              >
                Little Italy
              </option>
              <option
                value="Lower East Side"
              >
                Lower East Side
              </option>
              <option
                value="Marble Hill"
              >
                Marble Hill
              </option>
              <option
                value="Midtown East"
              >
                Midtown East
              </option>
              <option
                value="Midtown West"
              >
                Midtown West
              </option>
              <option
                value="Morningside Heights"
              >
                Morningside Heights
              </option>
              <option
                value="Murray Hill"
              >
                Murray Hill
              </option>
              <option
                value="NoHo"
              >
                NoHo
              </option>
              <option
                value="Nolita"
              >
                Nolita
              </option>
              <option
                value="Randall's Island"
              >
                Randall's Island
              </option>
              <option
                value="Roosevelt Island"
              >
                Roosevelt Island
              </option>
              <option
                value="SoHo"
              >
                SoHo
              </option>
              <option
                value="Spanish Harlem"
              >
                Spanish Harlem
              </option>
              <option
                value="Stuyvesant Town"
              >
                Stuyvesant Town
              </option>
              <option
                value="Tribeca"
              >
                Tribeca
              </option>
              <option
                value="Two Bridges"
              >
                Two Bridges
              </option>
              <option
                value="Upper East Side"
              >
                Upper East Side
              </option>
              <option
                value="Upper West Side"
              >
                Upper West Side
              </option>
              <option
                value="Washington Heights"
              >
                Washington Heights
              </option>
              <option
                value="West Village"
              >
                West Village
              </option>
            </select>
            <button
              className="btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="submit"
            >
              Add Tips
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
